lib/src/Uncertainty/Bayesian/NonLinearLeastSquaresCalibration.cxx

```diff
@@ -287,9 +287,30 @@ void NonLinearLeastSquaresCalibration::run()
   }
   else
   {
-    LinearLeastSquaresCalibration linearAlgo(model_, inputObservations_, outputObservations_, thetaStar);
-    linearAlgo.run();
-    parameterPosterior = linearAlgo.getResult().getParameterPosterior();
+    // Prepare the inverse Gram
+    // Compute the linearization
+    Function parametrizedModel(model_);
+    parametrizedModel.setParameter(thetaStar);
+    // Flatten everything related to the model evaluations over the input observations
```

Comment:
Isn't this a copy of the next code from lines 92 to 121 (with the exception of the diagonal regularization)?
https://github.com/openturns/openturns/blob/07a9b2174d4cfbf8382df26ed663e3402558299e/lib/src/Uncertainty/Bayesian/LinearLeastSquaresCalibration.cxx#L92
If yes, then the code can be factored, including the regularization.